### PR TITLE
Add build definitions for 67.1.

### PR DESCRIPTION
**Pull request by jpakkane**
Tested to work on Linux, Windows and Mac.

### CODE REVIEW

File: source/tools/toolutil/meson.build [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  'ppucd.cpp',
+  'flagparser.cpp',
+  'filetools.cpp',
+  'udbgutil.cpp',
+  'dbgutil.cpp',
+  'ucln_tu.cpp',
+)
+
+toolutil_c_args = ['-DU_TOOLUTIL_IMPLEMENTATION']
+toolutil_cpp_args = ['-DU_TOOLUTIL_IMPLEMENTATION']
```
Comment (eli-schwartz) [inline source/tools/toolutil/meson.build:29]: Hmm, any particular reason to define this twice? Looks like upstream rigidly defines all these *_IMPLEMENTATION defines regardless of language used, it's more like a toolutil_component_args.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#add_project_arguments('-DU_HAVE_STD_STRING=1', language : 'cpp')
+
+if host_machine.system() == 'windows'
+   add_project_arguments('-DU_PLATFORM_USES_ONLY_WIN32_API', language : 'cpp')
+   #add_project_arguments('-DU_PLATFORM_HAS_WIN32_API', language : 'cpp')
+   add_project_arguments('-DWIN32', '-DWIN64', '-D_MBCS', '/utf-8', language : 'cpp')
+   add_project_arguments('-DU_PLATFORM_USES_ONLY_WIN32_API', language : 'c')
+   #add_project_arguments('-DU_PLATFORM_HAS_WIN32_API', language : 'c')
+   add_project_arguments('-DWIN32', '-DWIN64', '-D_MBCS', '/utf-8', language : 'c')
+endif
```
Comment (eli-schwartz) [inline meson.build]: Could these be passed in once with `language: ['c', 'cpp']`?

What do the commented out defines do?

File: source/test/meson.build [hunk 3]
```diff
@@ -0,0 +1 @@
+#subdir('intltest')
```
Comment (eli-schwartz) [inline source/test/meson.build]: In comparison to 55.2 this is just dropped and commented out. Any particular reason?

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,32 @@
+project('icu', 'c', 'cpp',
+  version: '67.1',
+  meson_version: '>=0.55.99',
```
Comment (eli-schwartz) [inline meson.build]: Why is 0.55 not good enough anymore?

File: source/test/meson.build (hunk 3 above)
Comment (jpakkane) [reply source/test/meson.build]: It did not compile due to reasons I could not decipher.

File: source/tools/toolutil/meson.build (hunk 1 above)
Comment (jpakkane) [reply source/tools/toolutil/meson.build:29]: That's just how it was done generally. I think this was just preparing for the case where C and C++ might need different compiler flags. Dunno if that's going to come up.

File: meson.build (hunk 2 above)
Comment (jpakkane) [reply meson.build]: I took those from the VS project shipped by upstream (which did not actually compile, but I digress). Then I kept removing unnecessary ones until it worked. Commented out ones are deleted now.

File: meson.build (hunk 4 above)
Comment (jpakkane) [reply meson.build]: That is what I tested it on. Lowered to 0.55 as it should work on it, the main requirement is `[provides]`, but that was already in 0.55.

**Comment by jpakkane** [comment]
Does this still need work? I'd really want to get this in due to schwag PR reasons.

**Comment by eli-schwartz** [comment]
I finally figured out the reviewtool flow... If a wrap has "errors" such as the version not being in the upstream url due to upstream naming using weird munging rules, I need to pass an argument to ignore this so that it continues with the approval process.

**Pull request merged**